--- HList.pm	Fri Mar  6 01:33:34 1998
+++ HList.pm.patched	Fri Mar 20 11:20:44 1998
@@ -1,7 +1,7 @@
 package Tk::HList; 
 
 use vars qw($VERSION @ISA);
-$VERSION = '3.006'; # $Id: //depot/Tk8/HList/HList.pm#6$
+$VERSION = '3.006001'; # $Id: //depot/Tk8/HList/HList.pm#6$
 
 use Tk qw(Ev);
 
@@ -130,8 +130,7 @@
  if (defined($info[1]) && $info[1] eq 'indicator')
   {
    $w->{tixindicator} = $ent;
-   $w->EventType( "<Arm>" );
-   $w->Callback(-indicatorcmd => $ent);
+   $w->Callback(-indicatorcmd => $ent, "<Arm>");
   }
  else
   {
@@ -238,8 +237,7 @@
    my @info = $w->info('item',$Ev->x, $Ev->y);
    if(defined($info[1]) && $info[1] eq 'indicator')
     {
-     $w->EventType( "<Activate>" );
-     $w->Callback(-indicatorcmd => $ent);
+     $w->Callback(-indicatorcmd => $ent, "<Activate>");
     }
    return;
   }
@@ -293,8 +291,8 @@
 
  if($w->{tixindicator})
   {
-   $w->EventType( $w->{tixindicator} eq $ent ? "<Arm>" : "<Disarm>" );
-   $w->Callback(-indicatorcmd => $w->{tixindicator});
+   my $event_type = $w->{tixindicator} eq $ent ? "<Arm>" : "<Disarm>";
+   $w->Callback(-indicatorcmd => $w->{tixindicator}, $event_type );
    return;
   }
 
